lib/spinlock.h

```diff
@@ -22,7 +22,10 @@ extern "C" {
 /** \defgroup spinlock Spinlock Interfaces
  *  @{ */
 struct metal_spinlock {
-	atomic_int v;
+	union{
+		atomic_int v;
+		atomic_flag w;
+	};
```

Comment:
the atomic_flag is only available from c11, how about in the lib/atomic.h, we define atomic_flag if it is not defined.

And use atomic_flag only in the mutex and spinlock, as when checking the mutex and spinlock implementation in generic, it only uses flag.